src/dbnode/storage/index.go

```diff
@@ -1590,74 +1631,41 @@ func (i *nsIndex) queryWithSpan(
 			break
 		}
 
-		if applyTimeout := timeout > 0; !applyTimeout {
+		wg.Add(1)
```

Comment:
this is probably worth a close eye. it should just be a mechanical refactor to reduce the number of branches.